### DISCUSSION TITLE
Recommendations for Installing agent(s)

### DISCUSSION THREAD

**Discussion by lunabacilio** [community/community#27636] [Actions] [answered]
Hi everyone, 

Hope all you find well. 

I have a question regarding agents. I read the documentation and a couple of source about agents but I didn't find exactly the information about this: We can enable our self-hosted instances to use actions. So, What is the best practice of agents implementation? Install my server where I'm hosting my application(s) or enable a different instance to install and use it as agent and maintain both isolate? Talking about a production environment.

I will appreciate a lot your comments. 

Thanks.

**Comment by airtower-luna** [comment, marked as answer]
In general, you should *not* install CI runners (whether for GitHub Actions or other CI systems) into your production environment. If you want a workflow to deploy to your production system use appropriate deployment tools in the workflow to deploy *to* your production system (e.g. call Ansible with a deployment playbook), but don't deploy to the runner machine itself.

If you want to limit the deployment tool to running within your network (e.g. only allow SSH from your network, not outside) running the workflow on a self-hosted runner inside that network is a good option.

Hope that helps!

**Reply by lunabacilio** [reply]
This is the answer I was looking for. 

Thank you so much @airtower-luna.

**Comment by AbdelsameaHassan** [comment]
https://www.facebook.com/profile.php?id=100022235702075
